geth/params/config.go

```diff
@@ -205,6 +215,9 @@ type NodeConfig struct {
 	// LogToStderr defines whether logged info should also be output to os.Stderr
 	LogToStderr bool
 
+	// UpstreamConfig extra config for providing upstream infura server.
+	UpstreamConfig *UpstreamRPCConfig `json:"UpstreamConfig"`
```

Comment:
Am in agreement, but felt they used such approach for some certain reason I was not aware of. But if we are good we this, we can easily change this to a static struct.